src/VisualStudio/CSharp/Impl/CSharpPackage.cs

```diff
@@ -64,7 +63,7 @@ namespace Microsoft.VisualStudio.LanguageServices.CSharp.LanguageService
     [ProvideAutomationProperties("TextEditor", "CSharp-Specific", packageGuid: Guids.CSharpPackageIdString, profileNodeLabelId: 104, profileNodeDescriptionId: 105)]
     [ProvideService(typeof(CSharpLanguageService), ServiceName = "C# Language Service")]
     [ProvideService(typeof(ICSharpTempPECompilerService), ServiceName = "C# TempPE Compiler Service")]
-    internal class CSharpPackage : AbstractPackage<CSharpPackage, CSharpLanguageService, CSharpProjectShim>, IVsUserSettingsQuery
+    internal class CSharpPackage : AbstractPackage<CSharpPackage, CSharpLanguageService, AbstractProject>, IVsUserSettingsQuery
```

Comment:
The triple-generic-of-death pattern that exists here existed because we thought we needed that derived type. If this isn't necessary anymore, I'd be curious if the cleanup is easy or not to just remove the type argument entirely. In particular, AbstractPackage`3 exists just to derive from AbstractPackage2 right now!

If it's a huge ball of tangled string then just file a bug.